app/routes/joblistings/components/JoblistingEditor.tsx

```diff
@@ -1,342 +1,351 @@
+
+    if (isNew) {
+      await dispatch(createJoblisting(payload));
+    } else {
+      await dispatch(editJoblisting(payload));
+    }
+
+    const id = joblistingId || result.payload.result;
+    dispatch(push(`/joblistings/${id}/`));
   };
-  onDeleteJoblisting = () =>
-    this.props.deleteJoblisting(this.props.joblisting.id).then(() => {
-      this.props.push('/joblistings/');
+
+  const onDeleteJoblisting = () =>
+    dispatch(deleteJoblisting(joblistingId)).then(() => {
+      dispatch(push('/joblistings/'));
     });
-  fetchContacts = (company: SelectInputObject) => {
-    return this.props
-      .fetchCompanyContacts({
-        companyId: company.value,
-      })
-      .then((action) => {
+
+  const fetchContacts = useCallback(
+    (company: ListCompany) => {
+      return dispatch(
+        fetchCompanyContacts({
+          companyId: company.id,
+        })
+      ).then((action) => {
         const responsibleOptions = action.payload.map((contact) => ({
           label: contact.name,
           value: contact.id,
         }));
-        this.setState({
-          responsibleOptions,
-        });
+        setResponsibleOptions(responsibleOptions);
       });
-  };
+    },
+    [dispatch]
+  );
 
-  componentDidMount() {
-    if (!this.props.isNew) {
-      this.fetchContacts(this.props.company);
+  useEffect(() => {
+    if (!isNew && joblisting?.company) {
+      fetchContacts(joblisting?.company);
     }
-  }
+  }, [fetchContacts, isNew, joblisting]);
 
-  render() {
-    const {
-      handleSubmit,
-      isNew,
-      dispatch,
-      fetching = false,
-      submitting,
-      invalid,
-      push,
-    } = this.props;
+  const initialValues = {
+    ...joblisting,
+    text: joblisting?.text || '<p></p>',
+    description: joblisting?.description || '',
+    company: joblisting?.company
+      ? {
+          label: joblisting.company.name,
+          value: joblisting.company.id,
+        }
+      : {},
+    visibleFrom: joblisting?.visibleFrom || time({ hours: 12 }),
+    visibleTo:
+      joblisting?.visibleTo || time({ days: 31, hours: 23, minutes: 59 }),
+    deadline:
+      joblisting?.deadline || time({ days: 30, hours: 23, minutes: 59 }),
+    fromYear: yearValues.find(
+      ({ value }) => value === joblisting?.fromYear || value === 1
+    ),
+    toYear: yearValues.find(
+      ({ value }) => value === joblisting?.toYear || value === 5
+    ),
+    jobType: jobTypes.find(
+      ({ value }) => value === joblisting?.jobType || value === 'summer_job'
+    ),
+    responsible: joblisting?.responsible
+      ? {
+          label: joblisting.responsible.name,
+          value: joblisting.responsible.id,
+        }
+      : {
+          label: 'Ingen',
+          value: null,
+        },
+    workplaces: (joblisting?.workplaces || []).map((workplace) => ({
+      label: workplace.town,
+      value: workplace.town,
+    })),
+  };
 
-    if (!isNew && fetching) {
-      return <LoadingIndicator loading />;
-    }
```

Comment:
Is there any reason in particular you have deleted this? Isn't there still going to be stuff that might have to load?

Reply:
Yes, but if I remember correctly it looked better without. Most of the page does not have to wait